src/users/email_verifications/routes.rs

```diff
@@ -0,0 +1,64 @@
+use axum::extract::Path;
+use axum::routing::post;
+use axum::{extract::State, Router};
+use chrono::{Duration, Utc};
+use diesel::prelude::*;
+use diesel_async::{pooled_connection::deadpool::Pool, AsyncPgConnection, RunQueryDsl};
+use uuid::Uuid;
+
+use crate::schema::users;
+use crate::{auth::AuthExtractor, schema::email_verifications, users::models::UserRole, AppState};
+
+use super::{models::EmailVerification, EmailVerificationError};
+
+pub fn email_verification_router() -> Router<AppState> {
+    Router::new()
+        .route("/email_verification", post(create_email_verification))
+        .route("/confirm_email/:verification_id", post(confirm_email))
+}
+
+pub async fn create_email_verification(
+    auth: AuthExtractor<{ UserRole::User as u32 }>,
+    State(pool): State<Pool<AsyncPgConnection>>,
+) -> Result<(), EmailVerificationError> {
+    let mut db = pool.get().await?;
+    let email_verification = EmailVerification {
+        id: Uuid::now_v7(),
+        email: auth.current_user.email,
+        created_at: Utc::now(),
+        expires_at: Utc::now() + Duration::hours(1),
+        user_id: auth.current_user.id,
+    };
+    diesel::insert_into(email_verifications::table)
+        .values(&email_verification)
+        .execute(&mut db)
+        .await?;
+    email_verification
+        .send_email(auth.current_user.username)
+        .await?;
```

Comment:
isn't it better to get the resulting inserted item and then send the email with it? just for safety, and to make sure it's never sent unless the record was inserted 

docs:
https://docs.rs/diesel/latest/diesel/prelude/trait.RunQueryDsl.html#method.get_result

Reply:
I suppose that's true, but does it matter? since if it errors, it will propagate the error no?

Reply:
Yep doesn't matter